test/ufuzz.js

```diff
@@ -654,15 +654,18 @@ function _createSimpleBinaryExpr(recurmax, noComma, stmtDepth, canThrow) {
         var assignee = getVarName();
         var expr;
         switch (r) {
-          case 1:
-            return '(' + assignee + createAssignment() + s + ')';
+          case 3:
+            expr = '(' + assignee + '.' + getDotKey() + createAssignment() + s + ')';
+            break;
           case 2:
             expr = '(' + assignee + '[' + createExpression(recurmax, COMMA_OK, stmtDepth, canThrow)
                 + ']' + createAssignment() + s + ')';
-          case 3:
-            expr = '(' + assignee + '.' + getDotKey() + createAssignment() + s + ')';
+            break;
+          default:
+            expr = '(' + assignee + createAssignment() + s + ')';
+            break;
         }
-        return canThrow && rng(10) == 0 ? expr : '(' + assignee + ' && ' + expr + ')';
+        return r == 1 || canThrow && rng(10) == 0 ? expr : '(' + assignee + ' && ' + expr + ')';
```

Comment:
`r == 1` is the case of `variable = expr`, so we don't need any extra wrapping for that.